lib/sourcemap.js

```diff
@@ -144,7 +144,9 @@ function SourceMap(options) {
             add(source, gen_line, gen_col, orig_line, orig_col, name);
         } : add,
         setSourceContent: sources_content ? function(source, content) {
-            sources_content[source] = content;
+            if (!sources_content[source]) {
+                sources_content[source] = content;
+            }
```

Comment:
Admittedly an odd case, but this will still overwrite an existing _blank_ source file &minus; please consider the following instead:
```js
if (!(source in sources_content)) sources_content[source] = content;
```

Reply:
Sounds good!  For running on very old Node/browsers, would `if (!HOP(sources_content, source))` be better?  Or doesn't matter anymore and prefer readability at this point?

Reply:
`sources_content` is initiated to `Object.create(null)`, so in theory you won't get any inherited properties, i.e. in this particular case `source in sources_content` is equivalent to `HOP(sources_content, source)` on all platforms

Reply:
Ah, I missed the Object.create(null), thought this was a caller-provided map, you're right!  I've changed the check to `source in sources_content`.